File: Programming/Java/Palindrome/palindrome.java
package com.kholood;
import java.util.Scanner;
import java.lang.String;

public class palindrome {

    public static void main(String [] args){

            Scanner sc = new Scanner(System.in);
            System.out.println("Enter the number: ");
            String reverse = "";
            String num = sc.nextLine();
            int length = num.length();

            for( int i = length - 1; i >= 0; i-- )
                reverse = reverse + num.charAt(i);
            if (num.equals(reverse))
                System.out.println("The entered string " +num +" is a palindrome.");
            else
                System.out.println("The entered string " +num +"  isn't a palindrome.");

        System.out.print("Enter the string you want to check: ");
        String original = sc.nextLine();
        int n = original.length();
        String reversestring="";
        for(int i = n - 1; i >= 0; i--)
        {
            reversestring = reversestring + original.charAt(i);
        }
        if(original.equalsIgnoreCase(reversestring))
            System.out.println("The string is palindrome.");
        else
            System.out.println("The string is not palindrome.");


        }
    }

File: Programming/C/Palindrome/palindrome.c
#include<stdio.h>  // header file for input and output
 
 int palindrome(int n){
   int temp=n;   // temp variable to store entered value
   int change=0;

   if(n<0){
    return 0;
   }
   while(n>0){

    change =(change*10)+(n%10);      // change is to store reverse value
     n=n/10;
   }
   return (change==temp);      // checking the value you entered is same as reverse value if true then return 1
 }
int main(){
  int x;      // Number you want to check
  scanf("%d",&x);

  int result= palindrome(x);   // function to check for palindrome return true or false

  if(result==0){             // checking return value from palindrome function
    printf("false");         // false means not a palindrome
  }
  else
    printf("true");    // true means is a palindrome


 return 0;
}
